javascript/ql/lib/semmle/javascript/DOM.qll

```diff
@@ -421,6 +421,9 @@ module DOM {
     t.startInProp("target") and
     result = domEventSource()
     or
+    t.startInProp(DataFlow::PseudoProperties::arrayElement()) and
+    result = domElementCollection()
```

Comment:
Further up we use `domElementCollection()` directly as a `domValueSource()` (in the charpred for `DomValueSource::Range`).  

Should that be removed? Because it seems weird to have `domElementCollection()` both as direct DOM-value, and as something that contains a list of DOM-values.  

Otherwise LGTM.

Reply:
Unfortunately we don't yet have a precise enough distinction between collections and individual nodes. We'd need to first handle more things in `domElementCollection()`, things that are currently caught by the `getADomPropertyName()` catch-all.

I'm currently experimenting with translating the DOM model to data-extensions like we did for the SQL models, which should provide a much more robust model in general.